sale_stock_operating_unit/models/sale_order.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2015-19 Eficent Business and IT Consulting Services S.L. -
+# Jordi Ballester Alomar
+# Copyright 2015-19 Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import SUPERUSER_ID, _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.model
+    def _default_warehouse_id(self):
+        res = super(SaleOrder, self)._default_warehouse_id()
+        team = self._get_default_team()
+        warehouses = self.env["stock.warehouse"].search(
+            [
+                (
+                    "operating_unit_id",
+                    "=",
+                    team.with_user(SUPERUSER_ID).operating_unit_id.id,
+                )
+            ],
+            limit=1,
+        )
+        if warehouses:
+            return warehouses
+        return res
+
+    warehouse_id = fields.Many2one(
+        comodel_name="stock.warehouse", default=_default_warehouse_id
+    )
+
+    @api.onchange("team_id")
+    def onchange_team_id(self):
+        if self.team_id and self.team_id.operating_unit_id:
+            warehouses = self.env["stock.warehouse"].search(
+                [("operating_unit_id", "=", self.team_id.operating_unit_id.id)], limit=1
+            )
+            if warehouses:
+                self.warehouse_id = warehouses[0]
+
+    @api.onchange("operating_unit_id")
+    def onchange_operating_unit_id(self):
+        if self.operating_unit_id:
+            warehouses = self.env["stock.warehouse"].search(
+                [("operating_unit_id", "=", self.operating_unit_id.id)], limit=1
+            )
+            if warehouses:
+                self.warehouse_id = warehouses[0]
```

Comment:
```suggestion
    def onchange_operating_unit_id(self):
        if self.operating_unit_id:
            warehouses = self.env["stock.warehouse"].search(
                [("operating_unit_id", "=", self.operating_unit_id.id)]
            )
            if warehouses and self.warehouse_id not in warehouses:
                self.warehouse_id = warehouses[0]
```
Otherwise you can never change warehouse :). At some point I'll get this into a new PR